src/components/IGVBrowser.js

```diff
@@ -5,29 +5,36 @@ import { Link } from 'src/components/common'
 import { centeredSpinner, icon } from 'src/components/icons'
 import { saToken } from 'src/libs/ajax'
 import colors from 'src/libs/colors'
+import { getUserProjectForWorkspace, parseGsUri } from 'src/libs/data-utils'
 import { reportError } from 'src/libs/error'
+import { knownBucketRequesterPaysStatuses } from 'src/libs/state'
 import * as Utils from 'src/libs/utils'
 
 // format for selectedFiles prop: [{ filePath, indexFilePath } }]
-const IGVBrowser = ({ selectedFiles, refGenome, namespace, onDismiss }) => {
+const IGVBrowser = ({ selectedFiles, refGenome, workspace, onDismiss }) => {
   const containerRef = useRef()
   const [loadingIgv, setLoadingIgv] = useState(true)
 
   Utils.useOnMount(() => {
     const options = {
       genome: refGenome,
-      tracks: _.map(({ filePath, indexFilePath }) => ({
-        name: `${_.last(filePath.split('/'))} (${filePath})`,
-        url: filePath,
-        indexURL: indexFilePath
-      }), selectedFiles)
+      tracks: _.map(({ filePath, indexFilePath }) => {
+        const [bucket] = parseGsUri(filePath)
+        const userProjectParam = { userProject: knownBucketRequesterPaysStatuses.get()[bucket] ? getUserProjectForWorkspace(workspace) : undefined }
```

Comment:
Should we be prompting the user for the bucket if it is not in our list of known buckets?